src/nlu.cpp

```diff
@@ -297,3 +297,31 @@ std::string nlu::tokenize_str(std::string &input, std::string lang) {
   delete tokenizer_tmp;
   return to_return;
 }
+
+std::string nlu::detokenize_str(std::string &input) {
```

Comment:
Instead of having a complicated code (and difficult to maintain), we could just use a regex. Example:

```
  regex pattern(" ?' ?");
  string to_return = regex_replace(input, pattern, "'");
  return to_return;
```

Reply:
I'm used to avoid regex for complexity problem, but let's try it.